pmxbot/slack.py

```diff
@@ -115,16 +119,33 @@ def transmit(self, channel, message):
             If a ``thread`` attribute is present, that thread ID is used.
         :param str message: message to send.
         """
-        target = (
-            self.slack.server.channels.find(channel)
-            or self._find_user_channel(username=channel)
-            or self._find_user_channel(
-                username=self.get_email_username_map().get(channel)
-            )
-        )
         message = self._expand_references(message)
+        channel_id = self._get_channel_id(channel)
+        self.slack.web_client.chat_postMessage(
+            channel=channel_id, text=message, thread_ts=getattr(channel, 'thread', None)
+        )
+
+    @functools.lru_cache()
+    def _get_channel_id(self, channel):
+        # If this action was generated from a slack message event then we should have
+        # the channel_id already. For other cases we need to query the Slack API to get
+        # the channel id
+        if getattr(channel, "channel_id", None):
+            return channel.channel_id
 
-        target.send_message(message, thread=getattr(channel, 'thread', None))
+        channel_name = channel.strip('#')
+        cursor = None
+        while True:
+            resp = self.slacker.conversations.list(cursor=cursor, exclude_archived=True)
+            if not resp.successful:
+                log.error('Failed calls to conversations.list')
+                return None
```

Comment:
Is the right thing to do on an unsuccessful response to pretend like nothing happened? This `None` value will get cached. My instinct is this condition should trigger an Exception.

Reply:
I hadn't considered the caching of None, I agree with your approach, I will action.

Reply:
Using `self.slack.web_client.<endpoint>` to talk to slack does not return a `successful` field. Instead it will raise an exception, so I think this gives us what we want.